lib/disposable.dart

```diff
@@ -37,6 +37,43 @@ class _InternalDisposable implements _Disposable {
   }
 }
 
+/// A [Timer] implementation that exposes a [Future] that resolves when a
+/// non-periodic timer finishes it's callback or when any type of [Timer] is
+/// cancelled.
+class _ObservableTimer implements Timer {
+  Completer<Null> _didConclude = new Completer<Null>();
+  Timer _timer;
+
+  _ObservableTimer(Duration duration, void callback()) {
+    _timer = new Timer(duration, () {
+      callback();
+
+      if (!_didConclude.isCompleted) {
```

Comment:
Would we want to put the call to the callback in the guarded block as well? Presumably we would never want the callback to fire if the future has already completed. No?